integration-tests/src/test/resources/statedump.sh

```diff
@@ -60,16 +63,46 @@ function state_dump {
     done
   done
 
-  # use a job to archive PV, /scratch mounts to PV_ROOT in the K8S cluster
-  echo "Archiving pv directory using a kubernetes job.  Look for it on k8s cluster in $PV_ROOT/acceptance_test_pv_archive"
-  local outfile=${DUMP_DIR}/archive_pv_job.out
-  $SCRIPTPATH/job.sh "/scripts/archive.sh /scratch/acceptance_test_pv /scratch/acceptance_test_pv_archive" 2>&1 | tee ${outfile}
-  if [ "$?" = "0" ]; then
-     echo Job complete.
+
+  mkdir -p $ARCHIVE_DIR || fail Could not archive, could not create target directory \'$ARCHIVE_DIR\'.
+  
+  # use a run to archive PV, /sharedparent mounts to PV_ROOT in the K8S cluster
+  echo "Archiving pv directory using a kubernetes run.  Look for it on k8s cluster in $PV_ROOT/acceptance_test_pv_archive"
+  local outfile=${DUMP_DIR}/archive_pv_run.out
+  
+  if [ "$WERCKER" = "true" ]; then
+	$SCRIPTPATH/job.sh "/scripts/archive.sh /scratch/acceptance_test_pv /scratch/acceptance_test_pv_archive" 2>&1 | tee ${outfile}
+	if [ "$?" = "0" ]; then
+     	echo Job complete.
+  	else
+    	echo Job failed.  See ${outfile}.
+  	fi
   else
-     echo Job failed.  See ${outfile}.
+  
+  	$SCRIPTPATH/krun.sh -t 300 -d ${RESULT_DIR} -m "${PV_ROOT}:/sharedparent" -c 'jar cf /sharedparent/pvarchive.jar /sharedparent/acceptance_test_pv' 2>&1 | tee ${outfile}
+  	if [ "$?" = "0" ]; then
+    	$SCRIPTPATH/krun.sh -t 300 -d ${RESULT_DIR} -m  "${PV_ROOT}:/sharedparent" -c 'base64 /sharedparent/pvarchive.jar' > $RESULT_DIR/pvarchive.b64 2>&1
+	 	if [ "$?" = "0" ]; then
+   			base64 -di $RESULT_DIR/pvarchive.b64 > $ARCHIVE
+   			if [ "$?" = "0" ]; then
+   				echo Run complete. Archived to $ARCHIVE
+   			else 
+   				echo Run failed. 
+   			fi
+	 	else
+     		# command failed
+  			cat $RESULT_DIR/pvarchive.b64 | head -30
```

Comment:
maybe make this '-100' instead of '-30' - on an error the output may include a kubectl describe, and sometimes a kubetcl describe can be pretty big...

Reply:
Done